hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/balancer/TestBalancerService.java

```diff
@@ -210,4 +216,49 @@ public void testBalancerServiceOnError() throws Exception {
       }
     }
   }
+
+  @Test(timeout = 60000)
+  public void testBalancerServiceMetrics() throws Exception {
+    Configuration conf = new HdfsConfiguration();
+    conf.setTimeDuration(DFSConfigKeys.DFS_BALANCER_SERVICE_INTERVAL_KEY, 5, TimeUnit.SECONDS);
+    TestBalancer.initConf(conf);
+    try {
```

Comment:
It's best to use try-with-resources here.

Reply:
@tomscut Thanks for the review. The UT method of this class requires some additional operations on cluster, such as the `addOneDataNode()` method, which does not take cluster as a parameter.

If to change the usage, I can create a new JIRA to complete the task.

Reply:
Thanks for the explanation, there is no need to change here.